src/client/js/components/stock.js

```diff
@@ -0,0 +1,57 @@
+import React, { Component } from 'react';
+import request from 'ajax-request';
+
+export default class StockComponent extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      lastPrice: null,
+      dayLow: null,
+      dayHigh: null,
+      dayOpen: null,
+      change: null
+    }
+  }
+
+  componentDidMount() {
+    //Not sure how we're going to pass the stock info here
+    var nse = "GOOG";
+    stockObj = this.makeRequest(nse);
+    setStockState(stockObj);
+  }
+
+  makeRequest(nse) {
+    request({
+          url: `stock/${nse}`,
+          method: 'GET',
+          headers: {
+            'Content-Type':'application/x-www-form-urlencoded; charset=UTF-8'
+          }
+        }, (error, response, body) => {
+          const stockObj = JSON.parse(body);
+          return stockObj
```

Comment:
Without something like promises you cannot guarantee a return on a request. Instead why don't we just call `this.setStockState(stockObj)` right here (line 32) instead of expecting the return.